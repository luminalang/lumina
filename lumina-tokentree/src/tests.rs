use super::*;
use insta;

macro_rules! tree {
    ($name:ident, $src:literal) => {
        #[test]
        fn $name() {
            lumina_util::test_logger();
            let mut parser = Parser::new($src);
            let entity = parser.next(false);
            insta::assert_snapshot!(entity);
        }
    };
}

tree!(precedence, "1 = 2, 3 . 4 + 4 + 5 * 5, 6 + 6");
tree!(
    nested_match,
    "match 0
| outer0 -> r0
| outer1 ->
  match 1
  | inner0 -> 10
  | inner1 -> 11
| outer2 ->
  match 2
  | inner0 -> 20
  | inner1 -> 21
"
);
