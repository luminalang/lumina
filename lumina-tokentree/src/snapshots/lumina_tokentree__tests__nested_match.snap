---
source: lumina-tokentree/src/tests.rs
expression: entity
snapshot_kind: text
---
Header(match)
  Operators(3)
    0
  Op(0, |)
    [ outer0
    , Header(->)
        r0
    ]
  Op(1, |)
    [ outer1
    , Header(->)
        Header(match)
          Operators(1)
            1
          Op(0, |)
            Operators(1)
              [ inner0
              , Header(->)
                  10
              ]
            Op(0, |)
              [ inner1
              , Header(->)
                  11
              ]
    ]
  Op(2, |)
    [ outer2
    , Header(->)
        Header(match)
          Operators(1)
            2
          Op(0, |)
            Operators(1)
              [ inner0
              , Header(->)
                  20
              ]
            Op(0, |)
              [ inner1
              , Header(->)
                  21
              ]
    ]
